ethpm_types/abi.py

```diff
@@ -174,4 +174,42 @@ def signature(self) -> str:
         return f"{self.name}({input_args})"
 
 
-ABI = Union[ConstructorABI, FallbackABI, ReceiveABI, MethodABI, EventABI]
+class StructABIType(ABIType):
+    offset: int
+
+    @property
+    def signature(self) -> str:
+        sig = self.canonical_type
+        if self.name:
+            sig += f" {self.name}"
+
+        return f"{sig} {self.offset}"
+
+
+class StructABI(BaseModel):
+    type: Literal["struct"]
+
+    name: str
+    members: List[StructABIType]
```

Comment:
If `StructABIType` is not needed (or could be expanded w/ allowing extra fields), this could just be `ABIType`